### DISCUSSION TITLE
Marimo and Nebari

### DISCUSSION THREAD

**Discussion by pyinstallerbaggins** [nebari-dev/nebari#2705] [Q&A] [answered]
I was led here because I was using Jupyter notebooks looking to get the ability to control user access/permissions

But I am moving towards Marimo but I still like Nebari.

Can Marimo and Nebari pair together?

**Comment by Adam-D-Lewis** [comment, marked as answer]
Hi @pyinstallerbaggins, thanks for opening this issue.  I haven't seen Marimo before, but looking at it briefly it looks like Nebari couldn't support Marimo notebooks anytime soon.  The main reason is that Jupyterhub is a core component of Nebari and supports the concept of having multiple users, authentication, etc. There doesn't seem to be an equivalent of Jupyterhub in the Marimo ecosystem.

**Reply by pyinstallerbaggins** [reply]
Adam, thanks for the swift response!

**Comment by krassowski** [comment]
@pyinstallerbaggins reactive kernels such as https://github.com/ipyflow/ipyflow or https://github.com/davidbrochart/akernel can be used on nebari.

**Reply by pyinstallerbaggins** [reply]
thanks for the alternatives @krassowski !

I am leaning towards Marimo because it .py and not .ipynb, incredible upside for version control. Also, I can steer clear of conda, etc. going forward. Many other upsides too. 
 
The unfortunate thing is that Nebari is likely out of the wish list.

**Reply by Adam-D-Lewis** [reply]
@pyinstallerbaggins No worries!  One alternative I have seen people use is https://github.com/srstevenson/nb-clean#cleaning-pre-commit-hook which makes tracking notebooks is version control much easier.

**Reply by krassowski** [reply]
Also https://github.com/mwouts/jupytext with https://jupytext.readthedocs.io/en/latest/#collaborating-on-notebooks-with-git - we could probably include it by default in nebari docker images.
